### ISSUE TITLE
Game crashing after patching.

### ISSUE THREAD

**Issue by Maesejesus** [LeonarthCG/FFTA_Engine_Hacks#8]
First of all, I want to thank you, Leonarth, for the work you did. These improvements are really nice and useful! By far the No Stat Variance and the improvement on animations are my favourite. 

But I found a weird, game-breaking glitch, which I found after working on a modification of FFTA for quite some time and getting into test-playing it from beginning to end. Everything went fine at first, but I found out that during battles it seems there' a chance that the game hangs up when going through menus. More specifically I've seen this to happen when going in and out of the Items menu (if you have that command in a character). 

First I found it by accident playing when I "missclicked" the Item menu and wanted to go back. The game froze. I thought it was something of my doing, so then I tried with a new, clean ROM and applied your engine hacks and tested it again (only with your modifications) and it happened still. It doesn't happen every time, but if you keep going in and out of the Item menu, you're bound to make the game freeze. I tried to replicate this with a clean game but I did not see anything wrong. 

Sometimes the game doesn't outright freeze, but starts doing weirder stuff, like buffs suddenly appearing above character's heads, or text getting locked to the bottom of the screen in areas where the menu doesn't display text. After that it does freeze. Music plays fine, everything else stops. 

I wanted to ask if it's there any solution for this glitch, but maybe you no longer work on this anymore. Either way, thanks for your effort and have a nice day.

**Comment by LeonarthCG** [comment]
I have been trying to replicate this, but I've had no success... I would need more information, such as which engine hacks you installed, any modifications you did to the installation settings, which/how many items you had (or just a savefile, really), your choice of emulator, etc.
If you made any modifications with other tools, I'd also need information on that, or a patch that I can debug with.

Perhaps you could post about this in the ffhacktics forum, or join the discord server for FFTA hacking (also from ffhacktics)?

**Comment by Maesejesus** [comment]
Thank you so much for checking on it. Let me see...

I had the issue with any set of options I tried, this counts of course the default options the .event files have straight from being downloaded. For the hack I was working on, for example I didn't use anything related to the JP system (so no 1bitabilities and no custom names for example), but again, I tried with default settings and I got the same freeze.

About items, I actually didn't buy anything during my test playthrough, and I also got it to happen with an entire new save file (completely new .sav file too, just in case) going straight to the first battle and trying to replicate it, so the Items would be the default ones that come creating a new save file.

10 Potions
2 Phoenix Down
4 Bandage
4 Antidote

I replicated the problem on both VBA and mGBA. In fact I got it to freeze again while writing this. It seems to only happen with the Items command, I cannot make it happen with others. Sometimes before the freezing the Item list appears empty and freezes when you try to exit. 

I don't know what good would be to share the save file if it's an entirely new one, but here it is, alongside IPS files of what the resulting rom ends up being from your Engine Hack installer (with and without the JP system) from the last time I tried them. If needed, I could also share the patch of my current hack, although it's not been finished (as I need to testplay it to polish certain things). 

The ROM has a CRC32 of 5645E56C. Is that the correct one?

I considered posting first on the FFHacktics forum but I thought going here would be the correct way. 

[fftahack.zip](https://github.com/LeonarthCG/FFTA_Engine_Hacks/files/8343540/fftahack.zip)

**Comment by LeonarthCG** [comment]
Thanks a lot for letting me know about this issue, I was finally able to figure out what's going on.

Turns out when I moved the item list buffers it caused them to never be considered freed by the game, which would cause the game to crash after opening the item menu in battle about 27 times without resetting the game.

I just wrote and pushed a fix, please let me know if you are still able to reproduce the problem.

**Comment by Maesejesus** [comment]
Thank you for taking a look into it! 

It kinda checks out with the first time it crashed for me, as, while I tend to not use Items specially early on, I entered the Items menu several times throughout the around 2 hours of game time I did playing before entering the Items menu for the last time, causing the freeze. 

I tried it with the newer version and I was unable to replicate it, so it certainly looks you got it nailed! I'll keep trying just in case but seems to work perfectly now. 

Now I'll have to see how I can port my hack's changes into this new version (maybe if I repatch the modified ROM with the same FreeSpace offset?). In any case, again, thank you for fixing it and sorry for bothering you. I'll close this issue.

Have a nice day!
